bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java

```diff
@@ -162,10 +162,20 @@ public void dispose() {
         stopPolls();
         OAuthClientService oAuthService = this.oAuthService;
         if (oAuthService != null) {
-            oAuthService.close();
+            oAuthFactory.ungetOAuthService(getThing().toString());
```

Comment:
Perhaps also call `removeAccessTokenRefreshListener` here first? See:

https://github.com/openhab/openhab-addons/blob/5d65507584d6c66f457829d9e88d2be0864b3cfe/bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java#L598-L607

Reply:
Yes, I did not see that.

Reply:
Added